### DISCUSSION TITLE
getStaticProps vs. getServerSideProps for a blog index page?

### DISCUSSION THREAD

**Discussion by johanscreencloud** [vercel/next.js#24867] [Help] [answered]
Hi,

I'm building a blog page and I'm a bit confused about getStaticProps only getting called at build time. If I'm making a /blog page that should display all the latest blogposts (fetched from a CMS) I need to use getServerSideProps to fetch them right?

If I'd use getStaticProps, I would have to rebuild the server for any blog posts posted after the latest build to show up?

Thanks for any input!

**Comment by bobarros** [comment, marked as answer]
You still can use getStaticProps with revalidate:

```jsx
export async function getStaticProps() {
  const res = await fetch('https://.../posts')
  const posts = await res.json()

  return {
    props: {
      posts,
    },
    // Next.js will attempt to re-generate the page:
    // - When a request comes in
    // - At most once every second
    revalidate: 60, // In seconds
  }
}

export default Blog
```

More here:

> https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration

You only use getServerSideProps when you need some input from the user to build the page.

If revalidate was not an option, a webhook could rebuild your site still using getStaticProps, for example.

**Reply by johanscreencloud** [reply]
Thank you, makes sense! Think I'll go with ISR then 👍
